### PR TITLE
Plotly version 4.0.0 support

### PR DESCRIPTION
**Pull request by suecharo**
The current visualization was unable to render in google colab, etc..
This is because plotly ver3 installed by default.
Therefore, I modified plotly specify ver.4 as dependent libraries.
And modified plot_intermediate_values.

I did the following as test:

```
$ git clone <forked repo.>
$ cd optuna
$ docker run -it --rm -p 8888:8888 -v $PWD:/workdir -w /workdir python:slim bash
# pip3 install -U pip wheel setuptools
# pip3 install -e .
# pip3 list | grep plotly
plotly          4.1.1   
# pip3 install jupyter sklearn pytest
# jupyter notebook --ip 0.0.0.0 --port 8888 --allow-root

Run it.
http://localhost:8888/notebooks/examples/visualization/plot_intermediate_values.ipynb

# pytest tests/test_visualization.py
```

Output Example:

スクリーンショット 2019-09-11 18 20 26

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -40,8 +40,17 @@ def get_install_requires():
     # type: () -> List[str]
 
     install_requires = [
-        'sqlalchemy>=1.1.0', 'numpy', 'scipy', 'six',
-        'cliff', 'colorlog', 'pandas', 'alembic', 'typing'
+        'alembic',
+        'bokeh',
```
Comment (g-votte) [inline setup.py]: Please remove `bokeh` and `plotly` from `install_requires`.

File: optuna/visualization.py [hunk 2]
```diff
@@ -45,22 +45,47 @@ def objective(trial):
...
+        return go.Figure(data=[], layout=layout)
+
+    if 'intermediate_values' not in df:
+        return go.Figure(data=[], layout=layout)
+
+    target_state = [TrialState.PRUNED, TrialState.COMPLETE, TrialState.RUNNING]
+    dst_df = df[df['state'].isin(target_state)]
+    dst_df = dst_df[dst_df['intermediate_values'].isnull().all(axis=1) == False]
+    traces = []
+    for __, row in dst_df.iterrows():
```
Comment (g-votte) [inline optuna/visualization.py]: Please replace `__` with `_`.

File: optuna/visualization.py [hunk 3]
```diff
@@ -45,22 +45,47 @@ def objective(trial):
...
+    )
+
+    try:
+        df = study.trials_dataframe()
+        df = df.rename(columns={'number': 'trial_id'})
+    except Exception:  # empty study
+        return go.Figure(data=[], layout=layout)
+
+    if 'intermediate_values' not in df:
+        return go.Figure(data=[], layout=layout)
```
Comment (suecharo) [inline optuna/visualization.py]: should dump warning as like "this study doesn't have intermediate values"

File: optuna/visualization.py (hunk 2 above)
Comment (suecharo) [inline optuna/visualization.py]: Use one underscore as anonymous val.

File: optuna/visualization.py [hunk 4]
```diff
@@ -1,14 +1,18 @@
+from plotly.graph_objs._figure import Figure  # NOQA
```
Comment (sile) [inline optuna/visualization.py]: Please move this import to line 14.

**Comment by g-votte** [comment]
@suecharo 
Thanks for your contribution!

**Pull request merged**